packages/hash/frontend/src/blocks/page/BlockContextMenu/BlockContextMenu.tsx

```diff
@@ -43,178 +46,192 @@ const LOAD_BLOCK_ENTITY_UI = "hash-load-entity-ui";
 export const BlockContextMenu = forwardRef<
   HTMLDivElement,
   BlockContextMenuProps
->(({ popupState, blockSuggesterProps, entityId, entityStore }, ref) => {
-  const blockData = entityId ? entityStore.saved[entityId] : null;
-  const { data: users } = useUsers();
-  const setEntityMenuItemRef = useRef<HTMLLIElement>(null);
-  const swapBlocksMenuItemRef = useRef<HTMLLIElement>(null);
-
-  if (blockData && !isBlockEntity(blockData)) {
-    throw new Error("BlockContextMenu linked to non-block entity");
-  }
-
-  const menuItems = useMemo(() => {
-    const hasChildEntity =
-      Object.keys(blockData?.properties.entity?.properties ?? {}).length > 0;
-    const items = [
-      {
-        key: "set-entity",
-        title: hasChildEntity ? "Swap Entity" : "Add an entity",
-        icon: <FontAwesomeIcon icon={faAdd} />,
-        subMenu: <LoadEntityMenuContent entityId={entityId} />,
-        subMenuWidth: 280,
-      },
-      {
-        key: "copy-link",
-        title: "Copy Link",
-        icon: <FontAwesomeIcon icon={faLink} />,
-        onClick: () => {
-          const url = new URL(document.location.href);
-          url.hash = getBlockDomId(entityId!);
-          void navigator.clipboard.writeText(url.toString());
+>(
+  (
+    { blockData, popupState, blockSuggesterProps, entityId, openConfigMenu },
+    ref,
+  ) => {
```

Comment:
if you see a large diff below this line, you need to 'hide whitespace' and reload